src/ucs/config/parser.c

```diff
@@ -1885,10 +1902,16 @@ static void ucs_config_parser_warn_unused_env_vars(const char *prefix)
                  UCS_GLOBAL_OPTS_WARN_UNUSED_CONFIG);
     }
 
-    pthread_mutex_unlock(&ucs_config_parser_env_vars_hash_lock);
+    if (num_used_vars > 0) {
```

Comment:
maybe check if strb is empty?

Reply:
this counters also useful for string formatting (whether we have just one variable or several)